server/app/services/rpc_server.rb

```diff
@@ -93,15 +116,19 @@ def handling_actor(grid_id, name)
     @handlers[grid_id] ||= {}
     unless @handlers[grid_id][name]
       grid = Grid.find(grid_id)
-      @handlers[grid_id][name] = HANDLERS[name].new(grid) if grid
+      if grid
+        @handlers[grid_id][name] = HANDLERS[name].new(grid)
+      end
     end
 
     @handlers[grid_id][name]
   end
 
   # @param [Faye::Websocket] ws
-  # @param [Array] message
+  # @param [Object] message
   def send_message(ws, message)
-    ws.send(MessagePack.dump(message).bytes)
+    EM.next_tick {
```

Comment:
Oh, right. So the existing code was violating the EM threading, writing to the WS from the `RpcServer` actor thread? Hmm... :boom:

Perhaps this actor should be passed the `WebsocketBackend` instead, with some explicit comment about how `send_message` is okay for calling by other threads... mixing code for three different threads into the one Actor doesn't feel like a good idea.